spec.md

```diff
@@ -523,13 +518,295 @@ schema being used.
         "bullets": [
           "40 concurrent connections"
         ]
+      },
+      "schemas": {
+        "service_instance": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+            }
+          },
+          "update": {
+            "parameters": {
+              "$schema": "http://json-schema.org/draft-04/schema#",
+              "type": "object",
+              "propeties": {
+                "some-other-account": {
+                  "description": "Other fake account needed for update.",
+                  "type": "string"
+                },
+                "billing": {
+                  "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+                }
+              }
+            }
+          }
+        },
+        "service_binding": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1/bindings#AccountParameters"
+            }
+          }
+        }
       }
     }]
   }]
 }
 ```
 
 
+## JSON Schemas
+
+The JSON Schemas endpoint is used to define reusable definitions of JSON Schema
+Objects referenced throughout the [Catalog](#catalog-management). It will be the
+responsibility of the Platform to examine the response from the catalog to
+determine if the Broker is using `$ref` JSON Schemas. The Broker can signal to
+the Platform that JSON Schema components are supported by using the `osb_v2:`
+scheme. Components are a way to allow Platforms to only fetch sub-schemas.
+
+The root JSON Schema document can be found by getting the JSON Schemas
+endpoint without a `:component_path`.
+
+If the `osb_v2:` scheme is used by the broker, then the Platform MAY fetch the
+component JSON Schema directly using the following convention:
+
+Given: `osb_v2:///{:component_path}`, fetch `/v2/catalog/schemas/:component_path`
+and then the Platform is responsible to continue fetching any unknown
+`:component_path` URIs. Care ought to be taken by the platform to not fetch
+duplicate definitions.
+
+A Platform MAY ignore the `osb_v2:` scheme and just fetch `/v2/catalog/schemas`
+which will contain all sub-Schema documents. This reduces the complexity of the
+Platform but increases the potential payload size of the JSON Schemas.
+
+A Platform MUST fetch the root Schema document for any other `$ref` URIs found
+that do not match the `osb_v2:` scheme. If this happens the Platform has no use
+for the previously fetched sub-schemas and can just use the root schema. The
+Broker MUST provide the JSON Schema Object for the referenced URI in a subschema
+returned.
+
+Component Paths that use the scheme `osb_v2:` SHOULD be versioned, and the
+document behind the URI MUST NOT change after the catalog has been served to a
```

Comment:
I missed this MUST NOT before.  Does this include never deleting it? I hope not.

Reply:
Delete is fine unless it is still referenced by the catalog.

Reply:
I think this line addresses your concern with:

>However, the spec doesn't say anything about the "static-ness" of the data. For example, while it doesn't come right out and say it, I think there is a strong implication that the Platform will do a diff of the /v2/catalog results against the previous results to see what, if anything, has changed.

Correct?

Reply:
Saying component paths SHOULD be versioned feels awkward to me. It seems like it should be a MUST instead if the data can never be changed. But, do we really need to talk about versioning at all? That feels more like an impl detail/choice. What if we just said:
When the `osb_v2:` scheme is used, the data referenced by a component path MUST NOT change and can only be deleted when there are no longer any references to it in the Service Broker's schema. Additionally, once deleted, that component path MUST NOT ever be used again by this Service Broker.

Reply:
I like it. Replaced.